CodenameOne/src/com/codename1/io/JSONParser.java

```diff
@@ -632,4 +634,270 @@ public void keyValue(String key, String value) {
+     *
+     * Log.p("--- mapToJson() test");
+     * try {
+     *     Log.p("\n" + mapToJson(firstPerson));
+     * } catch (Exception ex) {
+     *     Log.e(ex);
+     * }
+     * }</pre>
+     *
+     * The output will be:
+     * <pre>{@code
+     * {
+     *   "firstName" : "Paco",
+     *   "lastName" : "Bellèz",
+     *   "isAlive" : true,
+     *   "age" : 35,
+     *   "weight (kg)" : 70.7,
+     *   "address" : {
+     *     "streetAddress" : "53, London Street",
+     *     "city" : "Paris",
+     *     "state" : "FR",
+     *     "postalCode" : "54856"
+     *   },
+     *   "partner" : {
+     *     "firstName" : "Gioia",
+     *     "lastName" : "Mia",
+     *     "isAlive" : true,
+     *     "age" : 34,
+     *     "weight (kg)" : 60.2,
+     *     "address" : {
+     *       "streetAddress" : "21 2nd Street",
+     *       "city" : "New York",
+     *       "state" : "NY",
+     *       "postalCode" : "10021-3100",
+     *       "phoneNumbers" : [{
+     *         "mobile" : "06124578965"
+     *       }]
+     *     }
+     *   },
+     *   "children" : [],
+     *   "extraInfo" : null,
+     *   "phoneNumbers" : [{
+     *     "home" : "212 555-1234"
+     *   },{
+     *     "office" : "646 555-4567"
+     *   },{
+     *     "mobile" : "123 456-7890"
+     *   }],
+     *   "onVacation" : false,
+     *   "friends" : ["Paul","Karl","Mary"]
+     * }
+     * }</pre>
+     *
+     * @param map The map to be converted to a JSON string
+     * @return The JSON string
+     * @throws Exception A generic exception thrown when converting the map to a
+     * JSON string fails.
+     * @author Francesco Galgani
+     */
+    public static String mapToJson(Map<String, Object> map) throws Exception {
+        String indent = "  ";
+        StringBuilder json = new StringBuilder();
+        json.append("{\n");
+        int remainingItems = map.size();
+        for (Map.Entry<String, Object> entry : map.entrySet()) {
+            json.append(indent).append("\"").append(entry.getKey()).append("\" : ");
+            if (entry.getValue() instanceof String) {
+                String value = (String) entry.getValue();
+                json.append("\"").append(value).append("\"");
+            } else if (entry.getValue() instanceof Number) {
+                String value;
+                if (entry.getValue() instanceof Byte) {
+                    value = ((Byte) entry.getValue()).toString();
+                } else if (entry.getValue() instanceof Double) {
+                    value = ((Double) entry.getValue()).toString();
+                } else if (entry.getValue() instanceof Float) {
+                    value = ((Float) entry.getValue()).toString();
+                } else if (entry.getValue() instanceof Integer) {
+                    value = ((Integer) entry.getValue()).toString();
+                } else if (entry.getValue() instanceof Long) {
+                    value = ((Long) entry.getValue()).toString();
```

Comment:
As JS has nothing but `Double` and therefore `Long` can't be represented exactly as number, it's common to offer an option serializing `Long` to `String`.